code-collab-app/src/editorpage.js

```diff
@@ -32,70 +30,68 @@ class Editorpage extends React.Component {
     if (!cellContents) {
       cellContents = '';
     }
-
-    this.setState({
-      cellText: this.state.cellText.concat(cellContents),
-      cellLocked: this.state.cellLocked.concat(false),
-    });
-
-    this.setState({
-      cellCount: this.state.cellCount + 1,
-    });
+    this.props.peer_data.add_new_cell(cellContents);
+    // TODO - maybe call peer_data method to update other peers
   }
 
   // Locks a cell so the user cannot edit it
   // key: the index of the cell to lock
   lockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = true;
-    this.setState({ cellLocked:temp });
+    // Lock cell so user may not edit
+    this.props.peer_data.set_cell_locked(key, true);
   }
 
-  // Unlocks a cell so the user may edit it
+  // Unlock a cell so the user may edit it
   // key: the index of the cell to unlock
   unlockEditorCell = (key) => {
-    let temp = this.state.cellLocked;
-    temp[key] = false;
-    this.setState({ cellLocked:temp });
+    this.props.peer_data.set_cell_locked(key, false);
+
   }
 
   // Requests lock on cell from all peers in session
   // key: index of cell
   requestEditorCellLock = (key) => {
     // TODO - implement socket.io request to lock this cell on all other peers in session
+    // Will probably involve calling lockEditorCell()
     console.log('Requesting cell lock from all peers for cell ' + key);
```

Comment:
Still todo on the peer logic side I presume? This is one of the events that will fire when a cell is selected, so it will need to call some peer logic to obtain a consensus lock from all peers before editing may occur.

Reply:
yes, this is still to come.